src/intersection-observer.js

```diff
@@ -81,20 +81,29 @@ export function getIntersectionChangeEntry(element, owner, viewport) {
       'Negative dimensions in element.');
   // Building an IntersectionObserverEntry.
 
-  let intersectionRect = element;
+  // The element is relative to (0, 0), while the viewport moves. So, we must
+  // adjust.
+  const boundingClientRect = moveLayoutRect(element, -viewport.left,
+      -viewport.top);
+
+  let intersectionRect = boundingClientRect;
   if (owner) {
-    intersectionRect = rectIntersection(owner, intersectionRect) ||
+    const ownerRect = moveLayoutRect(owner, -viewport.left, -viewport.top);
+    intersectionRect = rectIntersection(ownerRect, boundingClientRect) ||
         // No intersection.
         layoutRectLtwh(0, 0, 0, 0);
   }
-  intersectionRect = rectIntersection(viewport, intersectionRect) ||
+
+  // Now, a normal IntersectionObserver reports the viewport as (x,y) (0, 0).
+  const rootBounds = moveLayoutRect(viewport, -viewport.left, -viewport.top);
+  intersectionRect = rectIntersection(rootBounds, intersectionRect) ||
       // No intersection.
       layoutRectLtwh(0, 0, 0, 0);
 
   return /** @type {!IntersectionObserverEntry} */ ({
     time: Date.now(),
-    rootBounds: DomRectFromLayoutRect(viewport),
-    boundingClientRect: DomRectFromLayoutRect(element),
+    rootBounds: DomRectFromLayoutRect(rootBounds),
```

Comment:
I, sort of, can see both interpretation is being ok - since they look to be using the same coord system in both cases: one is adjusted to 0 and one is not. But the new system adds more calculations and object-creation. How does the spec define `rootBounds`? Obviously, I'd expect `boundingClientRect` to be viewport based...